Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py

```diff
@@ -0,0 +1,557 @@
+        :test_log_content: The contents of the test log as a string
+        :return: A dict of the tests and their respective Result objects
+        """
+        results = {}
+        pattern = re.compile(r"JSON_START\((.+?)\)JSON_END")
+        out_matches = pattern.finditer(output)
+        found_jsons = {}
+        for m in out_matches:
+            try:
+                elem = json.loads(m.groups()[0])
+                found_jsons[elem["name"]] = elem
+            except Exception:  # Intentionally broad to avoid failing if the output data is corrupt
+                continue
+
+        # Try to find the element in the log, this is used for cutting the log contents later
+        log_matches = pattern.finditer(test_log_content)
+        for m in log_matches:
+            try:
+                elem = json.loads(m.groups()[0])
+                if elem["name"] in found_jsons:
+                    found_jsons[elem["name"]]["log_match"] = m
+            except Exception:  # Intentionally broad, to avoid failing if the log data is corrupt
+                continue
+
+        log_start = 0
+        for test_spec in test_spec_list:
+            name = editor_utils.get_module_filename(test_spec.test_module)
+            if name not in found_jsons.keys():
+                results[test_spec.__name__] = Result.Unknown(
+                    test_spec, output,
+                    f"Found no test run information on stdout for {name} in the test log",
+                    test_log_content)
+            else:
+                result = None
+                json_result = found_jsons[name]
+                json_output = json_result["output"]
+
+                # Cut the editor log so it only has the output for this run
+                if "log_match" in json_result:
+                    m = json_result["log_match"]
+                    end = m.end() if test_spec != test_spec_list[-1] else -1
+                else:
+                    end = -1
+                cur_log = test_log_content[log_start: end]
+                log_start = end
+
+                if json_result["success"]:
+                    result = Result.Pass(test_spec, json_output, cur_log)
+                else:
+                    result = Result.Fail(test_spec, json_output, cur_log)
+                results[test_spec.__name__] = result
+
+        return results
+
+    @staticmethod
+    def _report_result(name: str, result: Result) -> None:
+        """
+        Fails the test if the test result is not a PASS, specifying the information
+        :name: Name of the test
+        :result: The Result object which denotes if the test passed or not
+        :return: None
+        """
+        if isinstance(result, Result.Pass):
+            output_str = f"Test {name}:\n{str(result)}"
+            print(output_str)
+        else:
+            error_str = f"Test {name}:\n{str(result)}"
+            pytest.fail(error_str)
+
+    def _setup_test_run(self,
```

Comment:
This will need to be moved to a subclass that does the actual setup (editor or material editor)